tools/istio-iptables/pkg/dependencies/implementation.go

```diff
@@ -82,15 +83,34 @@ func (r *RealDependencies) execute(cmd string, ignoreErrors bool, args ...string
 	return externalCommand.Run()
 }
 
-func (r *RealDependencies) executeXTables(cmd string, ignoreErrors bool, args ...string) error {
+func (r *RealDependencies) executeXTables(cmd string, ignoreErrors bool, args ...string) (err error) {
 	log.Infof("Running command: %s %s", cmd, strings.Join(args, " "))
 	externalCommand := exec.Command(cmd, args...)
 	stdout := &bytes.Buffer{}
 	stderr := &bytes.Buffer{}
 	externalCommand.Stdout = stdout
 	externalCommand.Stderr = stderr
 
-	err := externalCommand.Run()
+	for i := 0; i < 10; i++ {
+		stderr.Reset()
+		err = externalCommand.Run()
+		exitCode, ok := exitCode(err)
+		if !ok {
+			// cannot get exit code. consider this as non-retriable.
+			break
+		}
+
+		if !isXTablesLockError(stderr, exitCode) {
+			// Command succeeded, or failed not because of xtables lock.
+			break
+		}
+
+		// If command failed because xtables was locked, try the command again.
+		// Note we retry invoking iptables command explicitly instead of using the `-w` option of iptables,
+		// because as of iptables 1.6.x (version shipped with bionic), iptables-restore does not support `-w`.
```

Comment:
We are on 1.8 now in master branch, so for docker images we would potentially use it. But I guess that does not work for VMs and CNI which use host iptables.

Reply:
Ok sg. I can do that in a follow up.